description.md

```diff
@@ -0,0 +1,937 @@
+- Then, the extensions calldata follows correspondingly.
+
+Offsets contain a zero-based offset in calldata for each parameter and are packed as follows:
+
+| Parameter | Location, bytes |
+| --- | --- |
+| MakerAssetSuffix | [0..3] |
+| TakerAssetSuffix | [4..7] |
+| MakingAmountGetter | [8..11] |
+| TakingAmountGetter | [12..15] |
+| Predicate | [16..19] |
+| MakerPermit | [20..23] |
+| PreInteractionData | [24..27] |
+| PostInteractionData | [28..31] |
+
+The `CustomData` calldata is located after all extensions. Its start is defined as the end offset for `PostInteractionData`, and its end is the offset of the end of the calldata.
+
+**Example**
+
+For orders that have a predicate with a length of 120 bytes, a permit of 32 bytes included, and 32 bytes of custom data, the extension calldata should be packed as follows:
+
+| Offsets |  | MakerPermit offset | Predicate offset |  |  |  |  | Predicate calldata | Permit calldata | Custom data calldata |
+| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
+| [28..31] | [24..27] | [20..23] | [16..19] | [12..15] | [8..11] | [4..7] | [0..3] | [32..151] | [152..183] | [184..216] |
+| 152 | 152 | 152 | 120 | 0 | 0 | 0 | 0 | calldata (120 bytes) | calldata (32 bytes) | calldata (32 bytes) |
+
+The following is the final calldata which includes the offsets.
+
+```bash
+#  28-31    24-27    20-23    16-19    12-15     8-11      4-7      0-3
+00000098 00000098 00000098 00000078 00000000 00000000 00000000 00000000
+#         predicate end => ^^^^^^^^ ^^^^^^^^ <= predicate start
+#   permit end => ^^^^^^^^ ^^^^^^^^ <= permit start
+
+# followed by
+# 120 bytes of predicate calldata
+# 32  bytes of permit
+# 32  bytes of custom data
+```
+
+### **Non-ERC20 tokens swap**
+
+To swap tokens with a transfer function signature that do not comply with the IERC20 interface `IERC20.transferFrom(from, to, amount)`, you need to use a proxy contract. The proxy contract takes `from`, `to`, and `amount` arguments, maps them to the token transfer function, and adds additional parameters from the suffix calldata.
+
+In this case, `makerAsset` and/or `takerAsset` should contain the address of the proxy contract, and the token address should be encoded into the suffix with additional arguments. The suffix consists of packed bytes representing the calldata of parameters that are passed to the proxy contract.
+
+Examples of tokens that have already implemented this method are based on the following standards:
+
+- ERC721 - `IERC721.transferFrom(from, to, tokenId)`
+- ERC1155 - `IERC1155.safeTransferFrom(from, to, tokenId, amount, data)`
+
+To implement your custom proxy:
+
+1. Implement a function that receives `from`, `to`, and `amount` as the first three parameters, along with additional parameters that will be passed with `makerAssetSuffix` and/or `takerAssetSuffix`.
+2. Find the function name whose selector equals `transferFrom(from,to,amount)`.
+3. Implement `transferFrom` for the token.
```

Comment:
It looks like this items should be a little more detailed. Where the one should search function with specific selector? Does `transferFrom` from item 3 is equal to the method from item 2?

Reply:
fixed